model.rb

```diff
@@ -118,6 +118,20 @@ def redacted_columns
   end
 end
 
+module HealthMonitorMethods
```

Comment:
I'm not sure if this aligns with the Ruby programming philosophy, but we could add stub methods to prompt developers to implement them in the resources.

```ruby
    def init_health_monitor_session
      raise NoMethodError
    end

    def check_pulse(session:, previous_pulse:)
      raise NoMethodError
    end
```

Reply:
I don't know if Ruby philosophy says anything about this, but currently if we call `init_health_monitor_session` on a resource that doesn't implement it, we already get NoMethodError as this is what Ruby does by default.